modules/images/fetchpriority/load.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * Module Name: Fetchpriority
+ * Description: Adds `fetchpriority=high` parameter to the non lazy loaded images.
+ * Experimental: Yes
+ *
+ * @since n.e.x.t
+ * @package performance-lab
+ */
+
+/**
+ * Filter image tags in content to add fetchpriority to the image tag if lazy loaded is not set.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $filtered_image The filtered image.
+ * @param string $context        The context of the image.
+ *
+ * @return string image tag
+ */
+function fetchpriority_img_tag_add( $filtered_image, $context ) {
+
+	if ( 'the_content' !== $context ) {
+		return $filtered_image;
+	}
+
+	if ( strpos( $filtered_image, 'loading="lazy"' ) === false ) {
+		$filtered_image = str_replace( '<img ', '<img fetchpriority="high" ', $filtered_image );
+		remove_filter( 'wp_content_img_tag', 'fetchpriority_img_tag_add', 10, 2 );
+	}
+
+	return $filtered_image;
+}
+add_filter( 'wp_content_img_tag', 'fetchpriority_img_tag_add', 10, 2 );
```

Comment:
One other piece you need to add here is support for the featured image output. You can hook into the `post_thumbnail_html` filter a new callback that just wraps the other callback above. Then you can use the `the_post_thumbnail` context which is already used in WordPress core.

Please add a test for this function as well.
```suggestion
add_filter( 'wp_content_img_tag', 'fetchpriority_img_tag_add', 10, 2 );

/**
 * Filters the post thumbnail HTML to conditionally add the fetchpriority attribute.
 *
 * @since n.e.x.t
 *
 * @param string $html The post thumbnail HTML to filter.
 * @return string The filtered post thumbnail HTML.
 */
function fetchpriority_filter_post_thumbnail_html( $html ) {
	return fetchpriority_img_tag_add_attr( $html, 'the_post_thumbnail' );
}
add_filter( 'post_thumbnail_html', 'fetchpriority_filter_post_thumbnail_html' );
```

Reply:
this is in place in the codebase